src/main/java/net/minecraftforge/classloading/EnumValuesTransformer.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.classloading;
+
+import net.minecraft.launchwrapper.IClassTransformer;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.tree.ClassNode;
+import org.objectweb.asm.tree.FieldNode;
+
+import javax.annotation.Nullable;
+
+/**
+ * Transforms the field {@code private static final Enum[] $VALUES} in enums so
+ * that it is not considered final and got inlined.
+ */
+public class EnumValuesTransformer implements IClassTransformer, Opcodes
+{
+    @Override
+    public byte[] transform(String name, String transformedName, @Nullable byte[] basicClass)
+    {
+        if (basicClass == null)
+            return null;
+        ClassNode classNode = new ClassNode();
+        ClassReader classReader = new ClassReader(basicClass);
+        classReader.accept(classNode, 0);
```

Comment:
Oboy, creating a ClassNode for every class ever, maybe check the super class is Enum before doing anything in the transformer?

Reply:
Or even better use `ClassReader.getAccess` instead of `ClassNode.access` down bellow to avoid ClassNode creation on all classes.

Reply:
Heh I don't know how to effectively check enums. This is not slow: the old fluid field access transformers do the same thing.
https://github.com/MinecraftForge/MinecraftForge/blob/1.12.x/src/main/java/net/minecraftforge/fml/common/asm/transformers/EventSubscriptionTransformer.java
Even the event subscription transformer look at all classes. What else can I say?

Reply:
There is such a way? I'll see

Reply:
I'm not a core mod writer, so I don't know much about the details